nncf/onnx/quantization/quantize_model.py

```diff
@@ -251,4 +332,11 @@ def compress_weights_impl(
         advanced_parameters,
     )
     graph = NNCFGraphFactory.create(model)
-    return compression_algorithm.apply(model, graph, dataset=dataset)
+
+    compressed_model = compression_algorithm.apply(model, graph, dataset=dataset)
+
+    if external_data_dir:
+        remove_metadata(compressed_model, MetadataKey.EXTERNAL_DATA_DIR)
```

Comment:
Very fragile logic, since any algorithm inside the weight compression pipeline can make a copy.

Reply:
To resolve this, we can wrap the model in our custom class, but it requires significantly more changes to achieve the same result. Shall we discuss and choose a more optimal solution?

Reply:
Please add ```remove_metadata(model, MetadataKey.EXTERNAL_DATA_DIR)```

Reply:
Done